src/main/scala/com/github/unchama/seichiassist/subsystems/buildcount/subsystems/notification/System.scala

```diff
@@ -18,11 +24,18 @@ object System {
     buildCountReadAPI: BuildCountAPI[F, G, Player]
   ): F[A] = {
     val action: NotifyLevelUp[F, Player] = BukkitNotifyLevelUp[F]
+    val actionAmount: NotifyBuildAmountThreshold[F, Player] =
+      BukkitNotifyBuildAmountThreshold[F]
     StreamExtra.compileToRestartingStream("[buildcount.notification]") {
-      buildCountReadAPI.buildLevelUpdates.evalMap {
-        case (player, levelDiff) =>
-          action.ofBuildLevelTo(player)(levelDiff)
-      }
+      buildCountReadAPI
+        .buildAmountUpdateDiffs
+        .either(buildCountReadAPI.buildLevelUpdates)
+        .evalMap {
+          case Left((player, buildAmountDiff)) =>
+            actionAmount.ofBuildAmountTo(player)(buildAmountDiff)
+          case Right((player, levelDiff)) =>
+            action.ofBuildLevelTo(player)(levelDiff)
+        }
```

Comment:
`Stream` を `either` で混ぜなくとも、最後に合流させれば綺麗そうです（そうしたほうが複数の `Stream` に対して一般化しやすい）

```suggestion
      val levelNotification =
        buildCountReadAPI.buildLevelUpdates.evalMap {
          case (player, levelDiff) =>
            action.ofBuildLevelTo(player)(levelDiff)
        }

      val amountThresholdNotification =
        buildCountReadAPI.buildLevelUpdates.evalMap {
          case (player, amountDiff) =>
            actionAmount.ofBuildAmountTo(player)(amountDiff)
        }

      levelNotification.merge(amountThresholdNotification)
```